### ISSUE TITLE
IBM DataStax DSE 6.8: gocql: unsupported protocol response version: 72

### ISSUE THREAD

**Issue by MiloSram** [apache/cassandra-gocql-driver#1604]
### What version of Cassandra are you using?
IBM DataStax DSE 6.8
[cqlsh 6.8.0 | DSE 6.8.15 | CQL spec 3.4.5 | DSE protocol v2]

### What version of Gocql are you using?
github.com/gocql/gocql v0.0.0-20220224095938-0eacd3183625

### What version of Go are you using?
go 1.17

### What did you do?
Following example https://github.com/ErickRamirezAU/astra_gocql_connect/blob/master/astra_gocql_connect.go
with public bundle and credentials (downloaded from IBM DataStax config page), which are working using Python cqlsh 6.8.0  from my local host and connecting without problem to Cassandra.

### What did you expect to see?
successful connection to db

### What did you see instead?
gocql: unable to create session: unable to discover protocol version: gocql: unsupported protocol response version: 72

**Comment by martin-sucha** [comment]
It seems that the server replied with a CQL protocol version that gocql does not support. Maybe 72 means `DSE protocol v2`?

As a workaround, you can try to specify

```go
cluster.ProtoVersion = 4
```

to force connecting with CQL protocol version 4.

**Comment by martin-sucha** [comment]
It seems DSE protocols use 65 and 66:

https://github.com/datastax/python-driver/blob/15d715f4e686032b02ce785eca1d176d2b25e32b/cassandra/__init__.py#L172-L180

So you could also try to capture a network dump with wireshark/tcpdump and check if the version field is really 72 or if gocql is interpreting some other part of the response as protocol version.

**Comment by MiloSram** [comment]
Python works with DSE v2, but for Grafana plugin backend I need to use go.
I'm setting:
	_cluster.ProtoVersion = 4
	_cluster.CQLVersion = "3.4.5"
	_cluster.SslOpts.MinVersion = tls.VersionTLS12
	_cluster.ConnectTimeout = time.Second * 10

and also (although some are apparently redundant):

[connection]
hostname = xxxxxxxxxxxxxrg.databases.appdomain.cloud
port = 30933
ssl = true

[protocol]
version=4 
dse_version = 0x42


Not specifying cluster.ProtoVersion I get the error just once:
gocql: unable to create session: unable to discover protocol version: gocql: unsupported protocol response version: 72
If I specify it, I get the error for each host in the cluster.

**Comment by martin-sucha** [comment]
Hmm, so far it seems the server responds with a protocol version that gocql does not support or that the response format is unexpected. I couldn't find 72 to be any protocol version.

It seems there is `native_transport_allow_older_protocols` option on the server side or `nodetool disableolderprotocolversions` command. Are older protocol versions disabled by any chance? (https://issues.apache.org/jira/browse/CASSANDRA-14659) gocql does not support cql protocol version 5 yet.

I don't have access to DSE to reproduce. Could you please capture the network traffic between the client and server, plaintext or decrypt the TLS stream (with SSLKEYLOGFILE/TLSConfig.KeyLogWriter that e.g. Wireshark supports) and share the plaintext packet contents as a pcap file? (Of course, please make sure the comms don't contain secrets like username+password that shouldn't be shared, e.g. by configuring some dummy values in the client configuration)

**Comment by ErickRamirezAU** [comment]
> IBM DataStax DSE 6.8 [cqlsh 6.8.0 | DSE 6.8.15 | CQL spec 3.4.5 | DSE protocol v2]

Why are you calling it "IBM DataStax"? IBM doesn't own DataStax or the DataStax Enterprise product. 🙂

**Comment by OleksiienkoMykyta** [comment]
> Python works with DSE v2, but for Grafana plugin backend I need to use go. I'm setting: _cluster.ProtoVersion = 4 _cluster.CQLVersion = "3.4.5" _cluster.SslOpts.MinVersion = tls.VersionTLS12 _cluster.ConnectTimeout = time.Second * 10
> 
> and also (although some are apparently redundant):
> 
> [connection] hostname = xxxxxxxxxxxxxrg.databases.appdomain.cloud port = 30933 ssl = true
> 
> [protocol] version=4 dse_version = 0x42
> 
> Not specifying cluster.ProtoVersion I get the error just once: gocql: unable to create session: unable to discover protocol version: gocql: unsupported protocol response version: 72 If I specify it, I get the error for each host in the cluster.

It's a little bit strange to me, you provide config from Python driver, but logs from gocql, seem inconsistent to me.
Anyway, gocql supports proto v5 now(63b6d7830710a0c6c5411c08bf16d38cb88d2a8a), you can try it, but not sure that it will help. 
However, the gocql driver does not natively support DataStax Enterprise (DSE) protocol version 2, see the [readme](https://github.com/apache/cassandra-gocql-driver?tab=readme-ov-file#readme). 
@joao-r-reis, should we support it, or it's something to be closed as not planed?

**Comment by joao-r-reis** [comment]
I don't think we should consider supporting DSE protocol version right now. At the very least this would require a wider community discussion about support of proprietary protocols in general... Closing it.
